lib/rules/no-object-constructor.js

```diff
@@ -80,6 +81,72 @@ module.exports = {
             return false;
         }
 
+        /**
+         * Determines whether a parenthesized object literal that replaces a specified node needs to be preceded by a semicolon.
+         * @param {ASTNode} node The node to be replaced. This node should be at the start of an `ExpressionStatement` or at the start of the body of an `ArrowFunctionExpression`.
+         * @returns {boolean} Whether a semicolon is required before the parenthesized object literal.
+         */
+        function needsSemicolon(node) {
+            const prevToken = sourceCode.getTokenBefore(node);
+
+            if (
+                !prevToken ||
+                prevToken.type === "Punctuator" && [":", ";", ">", "{", "=>", "++", "--"].includes(prevToken.value)
```

Comment:
Because this array never changes, I'd suggest moving it out of this function into a constant and using a set instead of an array. The set `has()` method is more efficient than the array `includes()` method.